src/vs/workbench/services/languageRuntime/common/languageRuntimeDataExplorerClient.ts

```diff
@@ -244,6 +251,11 @@ export class DataExplorerClientInstance extends Disposable {
 				this._asyncTasks.delete(e.callback_id);
 			}
 		}));
+
+		// Initialize the guessed syntax
+		this.guessCodeSyntax().then(syntax => {
+			this.guessedSyntax = syntax;
+		});
```

Comment:
Possibly paranoid, but should this check that the suggested (guessed, in the current vocabulary) syntax is among the ones that this data explorer instance claims to support? Given that syntax can't be an actual enum on this side, it feels like we should be a bit pedantic re: checking that the backend is internally consistent.

Reply:
Added early erroring/returns if 1) the backend has stated it does not support any syntaxes to convert or 2) if the desired syntax is not one of the options available. This definitely feels a lot safer!